src/Features/Core/Portable/Completion/Providers/AbstractSymbolCompletionProvider.cs

```diff
@@ -419,6 +419,12 @@ private Task<SyntaxContext> GetOrCreateContext(Document document, int position,
         {
             lock (s_cacheGate)
             {
+                // Make sure the cached task isn't cancelled to avoid endlessly returning a cancelled task.
+                if (s_cachedDocuments.TryGetValue(document, out var cachedContext) && cachedContext.IsCanceled)
```

Comment:
This isn't correct. Consider the scenario where two requests happen with different cancellation tokens, and then the first one (which is in the cache) cancels. The second didn't cancel, but got their task cancelled on them.

This is what AsyncLazy is for.